### ISSUE TITLE
cannot build with Visual Studio 2017

### ISSUE THREAD

**Issue by helo-ch** [eclipse-paho/paho.mqtt.c#370]
I am trying to build the library on Windows using Visual Studio 2017, but I get a lot of errors telling me some files couldn't be opened : 

`fatal error C1083: Impossible d'ouvrir le fichier include : 'openssl/ssl.h' : No such file or directory`

`fatal error C1083: Impossible d'ouvrir le fichier include : 'VersionInfo.h' : No such file or directory`

`fatal error C1083: Impossible d'ouvrir le fichier include : 'openssl/applink.c' : No such file or directory`

It seems all of the errors I'm getting are from these 3 files. 

I also get error LNK1181 messages telling me `unable to open paho-mqtt3a.lib` and `unable to open paho-mqtt3c.lib`

Is there any other way to build the library on Windows? 

I'm attaching a file containing all the error messages I'm getting in case I forgot to mention something relevant.
[all errors.txt](https://github.com/eclipse/paho.mqtt.c/files/1482392/all.errors.txt)

**Comment by icraggs** [comment]
Have you installed OpenSSL?  Are you using CMake to generate the Visual Studio files?  If you don't need to build with OpenSSL you can omit it by setting PAHO_WITH_SSL to false if using CMake.

**Comment by helo-ch** [comment]
Hi, thanks for the quick answer.

To be honest I don't know if OpenSSL is installed on the computer I was working on. What happened exactly is that I tried to build the library using the msbuild command as instructed [here](https://www.eclipse.org/paho/clients/c/), but I kept getting errors telling me the v141 build tools were missing. Looking it up on the internet I saw those tools were part on Visual Studio 2017, so I guessed I needed it to build the library, and a collegue lent me his computer with VS2017 installed so that I could build the library. I am not generating any file, only trying to build the library from the files sitting in the "Windows Build" repository of the library. 
I probably don't need to build with open SSL, but isn't the "VersionInfo.h" file still going to pose a problem?

**Comment by gkiryaziev** [comment]
You have to use CMake for generating project for your VS2017, this is how I do it for my VS2015, maybe it will help you too:

1. download and install CMake
2. download and unpack library sources
3. go to the unpacked folder and make new folder `build`
4. run cmake-gui and select `sorce folder` and `build` folder, click `Configure` button
5. select your VS2017 compiler

![2017-11-19_112546](https://user-images.githubusercontent.com/8873218/32989520-8039bf78-cd20-11e7-8690-efbebc0c4a95.png)

6. here is my settings

![2017-11-19_112754](https://user-images.githubusercontent.com/8873218/32989603-2d27b61c-cd22-11e7-9cd5-aa824a78733b.png)

7. click Configure - Generate - Open Project
8. build `INSTALL` project

![2017-11-19_120210](https://user-images.githubusercontent.com/8873218/32989579-bb138560-cd21-11e7-8d21-a6e9fbd8c1ac.png)

9. find in the install folder your files, `bin`, `include`, `lib`

![2017-11-19_115237](https://user-images.githubusercontent.com/8873218/32989522-8b4f691c-cd20-11e7-9479-3a1060485968.png)

**Comment by helo-ch** [comment]
Thanks for the detailed answer.

I tried that, but I got errors : `Error in configuration process, project files may be invalid`

Here is my setting and the errors I get : 

![capture](https://user-images.githubusercontent.com/28846165/33010387-97bbdfea-cdda-11e7-935c-777d70a72024.PNG)

I checked the CMakeFile, the missing destinations seem to be given, but I am new to CMake so I don't know if I am missing something. 
The CMakeOutput.log file doesn't seem to have any relevant message, there is nothing in it about any error.
